r/tools/linuxlibs.R

```diff
@@ -0,0 +1,341 @@
+      libfile <- NULL
+    }
+  } else {
+    libfile <- NULL
+  }
+  libfile
+}
+
+# Function to figure out which flavor of binary we should download.
+# By default, it will try to discover from the OS what distro-version we're on
+# but you can override this by setting the env var LIBARROW_BINARY_DISTRO to:
+# * `FALSE` (not case-sensitive), which tells the script not to download a binary,
+# * some other string, presumably a related "distro-version" that has binaries
+#   built that work for your OS
+identify_os <- function(os = Sys.getenv("LIBARROW_BINARY_DISTRO")) {
+  if (nzchar(os)) {
+    if (identical(tolower(os), "false")) {
+      # Env var says not to download a binary
+      return(NULL)
+    } else {
+      # Env var provided an os-version to use--maybe you're on Ubuntu 18.10 but
+      # we only build for 18.04 and that's fine--so use what the user set
+      return(os)
+    }
+  }
+
+  if (nzchar(Sys.which("lsb_release"))) {
+    distro <- tolower(system("lsb_release -is", intern = TRUE))
+    os_version <- system("lsb_release -rs", intern = TRUE)
+    # In the future, we may be able to do some mapping of distro-versions to
+    # versions we built for, since there's no way we'll build for everything.
+    os <- paste0(distro, "-", os_version)
+  } else if (file.exists("/etc/system-release")) {
+    # Something like "CentOS Linux release 7.7.1908 (Core)"
+    system_release <- tolower(utils::head(readLines("/etc/system-release"), 1))
+    # Extract from that the distro and the major version number
+    os <- sub("^([a-z]+) .* ([0-9]+).*$", "\\1-\\2", system_release)
+  } else {
+    cat("*** Unable to identify current OS/version\n")
+    os <- NULL
+  }
+  os
+}
+
+download_source <- function() {
+  tf1 <- tempfile()
+  src_dir <- NULL
+  source_url <- paste0(arrow_repo, "src/arrow-", VERSION, ".zip")
+  try(
+    download.file(source_url, tf1, quiet = quietly),
+    silent = quietly
+  )
+  if (!file.exists(tf1)) {
+    # Try for an official release
+    try(
+      download.file(apache_src_url, tf1, quiet = quietly),
+      silent = quietly
+    )
+  }
+  if (file.exists(tf1)) {
+    cat("*** Successfully retrieved C++ source\n")
+    src_dir <- tempfile()
+    unzip(tf1, exdir = src_dir)
+    unlink(tf1)
+    # These scripts need to be executable
+    system(sprintf("chmod 755 %s/cpp/build-support/*.sh", src_dir))
+    options(.arrow.cleanup = c(getOption(".arrow.cleanup"), src_dir))
+    # The actual src is in cpp
+    src_dir <- paste0(src_dir, "/cpp")
+  }
+  src_dir
+}
+
+find_local_source <- function() {
+  if (file.exists("../cpp/src/arrow/api.h")) {
+    # We're in a git checkout of arrow, so we can build it
+    cat("*** Found local C++ source\n")
+    return("../cpp")
+  } else {
+    return(NULL)
+  }
+}
+
+build_libarrow <- function(src_dir, dst_dir) {
+  # We'll need to compile R bindings with these libs, so delete any .o files
+  system("rm src/*.o", ignore.stdout = quietly, ignore.stderr = quietly)
```

Comment:
Invoke `make clean`, it does more than this.

Reply:
I don't need the other things `make clean` does. Removing the `.o` files is enough to get R to rebuild everything.